### DISCUSSION TITLE
Browser Extension: Streamline Autofill UX: Direct Autofill Option for Single Entries

### DISCUSSION THREAD

**Discussion by invertedEcho** [bitwarden/clients#7996] [Password Manager]
### Code Contribution Proposal

When right-clicking an input field and selecting Bitwarden's autofill, a submenu appears regardless of the number of available autofill entries. This forces users to navigate through an extra-step even for a single entry.

Proposed change:

- If only one entry is available, show the name and e-mail of the entry next to the current options, instead of showing a submenu

This would benefit users by reducing the clicks, saving time and improving user experience.

Considerations:

Some users might not like this. A new option in the extension's settings could be introcuded, that would allow users to toggle this.

**Comment by alide123321** [comment]
I filed a request back in 2022 when I was just starting to learn to code. I've learned enough since then, and if this gets approved, I'd like to take a shot at implementing it myself as my first open-source contribution.
[Original post](https://community.bitwarden.com/t/skip-the-auto-fill-tab-when-right-clicked-on-the-browser-if-there-is-only-one-option/37653)

Since this one is closed, I opened a new one: https://github.com/orgs/bitwarden/discussions/21436
